sandbox_dns_unix.go

```diff
@@ -184,6 +187,13 @@ func (sb *sandbox) setupDNS() error {
 			return err
 		}
 	} else {
+		// If embedded DNS server is enabled later for this contianer ipv4 loopback nameservers if
+		// present should be used. retain it before filtering out resolv.conf contents.
```

Comment:
`With this change, we filter out the 127.0.0.1 from the docker0 default network , but retaining it for user-defined network and handling it in host namespace.`

Yes, for user defined networks we have a way to handle the loopback nameserver in the host namespace. Hence it is ok to not filter it. But for default bridge network, 127.0.0.1 will still be the container's loopback. So filtering it out is the right behavior.